ckanext/dia/harvester/csw.py

```diff
@@ -206,6 +230,74 @@ def get_package_dict(self, context, data_dict):
+        if spatial_srid and spatial_geojson is not None:
+            if cornerPoints:
+                    log.warn('Multiple geometries, with a single SRID. Will not convert Geometry. cornerPoints:  {} spatial_geojson: {} SRID: EPSG:{}'.format(cornerPoints, spatial_geojson, spatial_srid))
+            else:
+                from pyproj import Proj, transform
+
+                outProj = Proj('+init=EPSG:4326')
+                inProj = Proj('+init=EPSG:' + spatial_srid)
+
+                if spatial_geojson['type'] == 'MultiPolygon':
+                    new_polygons = []
+                    for polygon in spatial_geojson['coordinates']:
+                        new_linestrings = []
+                        for linestring in polygon:
+                            new_linestring = []
+                            for x,y in linestring:
+                                nx, ny = transform(inProj, outProj, x, y)
+                                new_linestring.append([nx, ny])
+                            new_linestrings.append(new_linestring)
+                        new_polygons.append(new_linestrings)
+                    spatial_geojson['coordinates'] = new_polygons
+
+                elif spatial_geojson['type'] == 'Polygon':
+                    new_linestrings = []
+                    for linestring in spatial_geojson['coordinates']:
+                        new_linestring = []
+                        for x,y in linestring:
+                            nx, ny = transform(inProj, outProj, x, y)
+                            new_linestring.append([nx, ny])
+                        new_linestrings.append(new_linestring)
+                    spatial_geojson['coordinates'] = new_linestrings
+
+                elif spatial_geojson['type'] == 'Point':
+                    # {"type": "Point", "coordinates": [2145000.0, 5467000.0]}'}
+                    x,y = spatial_geojson['coordinates']
+                    nx, ny = transform(inProj, outProj, x, y)
+                    spatial_geojson['coordinates'] = [nx, ny]
+                else:
+                    msg = 'The DIA CSW harvest does not understand how to re-project a {} type of geojson'
+                    log.warn(msg.format(spatial_geojson['spatial']))
+
+            # create updated version of extras array with correct SRID + spatial field
+            new_extras = []
+            for extra in package_dict['extras']:
+                if extra['key'] == 'spatial':
+                    new_extras.append({'key': 'spatial', 'value': json.dumps(spatial_geojson)})
+                elif extra['key'] == 'spatial-reference-system':
+                    new_extras.append({'key': 'spatial-reference-system', 'value': '4326'})
+                else:
+                    new_extras.append(extra)
+            package_dict['extras'] = new_extras
+        else:
+            log.warning('Could not determine SRID or spatial field for package {}'.format(package_dict))
```

Comment:
Seems like we could refactor this. logic, as there are a lot of branches to content with at the moment. But I'm largely happy with the logic.